### PR TITLE
[dashboard] Refactoring of cache headers

### LINKED ISSUE #3818: [proxy] Make sure we do not (long-term) cache error responses
We recently introduced long-term caching for [static assets](https://github.com/gitpod-io/gitpod/blob/e56ed09ccb99dccd27507c3559ca64a49e857786/chart/config/proxy/lib.locations.conf#L25-L32).

While atm we have only a handful of resources under `/static` we should make sure that we do not accidently cache errors like 404s in the future, for instance by using a pattern like this: https://serverfault.com/a/690258

### PR DESCRIPTION
**Pull request by aledbf**
fixes #3818

### CODE REVIEW

File: components/dashboard/conf/Caddyfile [hunk 1]
```diff
@@ -7,29 +7,38 @@
...
+	handle @static {
+		header Cache-Control "public, max-age=31536000"
+	}
+
+	handle @notstatic {
+		header Cache-Control "no-cache, no-transform, must-revalidate"
+	}
+
+	header -Server
+	header -Last-Modified
```
Comment (geropl) [inline components/dashboard/conf/Caddyfile:40]: nit: this could be moved to `@notstatic`. But does not harm here, either.

**Review by akosyakov — APPROVED** [review]
cache-control headers looking good, static resources are not fetched on reload, but non static are always checked

File: components/dashboard/conf/Caddyfile [hunk 2]
```diff
@@ -7,29 +7,38 @@
...
 
 	redir /.well-known/security.txt https://www.gitpod.io/.well-known/security.txt permanent
 	redir /environment-variables /settings/ permanent
 
+	handle @static {
+		header Cache-Control "public, max-age=31536000"
+	}
+
+	handle @notstatic {
+		header Cache-Control "no-cache, no-transform, must-revalidate"
```
Comment (geropl) [inline components/dashboard/conf/Caddyfile:36]: I'd expect at least a short comment about the two types of content and about the intent and trade-offs regarding the two caching strategies.

**Review by geropl — APPROVED** [review]
> fixes #3818

I'm not sure it does. What is the default behavior of caddy on error responses?

I'd love to see more comments here, but let's merge now as it:
 - works as the current setup
 - unblocks a hotfix deployment

File: components/dashboard/conf/Caddyfile [hunk 3]
```diff
@@ -7,29 +7,38 @@
...
-
 (discard_log) {
 	log {
 		output discard
 	}
 }
 
 :80 {
-	import caching
+	@static {
```
Comment (akosyakov) [inline components/dashboard/conf/Caddyfile:17]: Does it ensure that 404 won't be cached? Should we do something like here https://caddy.community/t/correct-way-to-set-expires-on-caddy-2/7914/13?

**Pull request merged**